x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/async/GetAsyncStatusRequestTests.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.async;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+import static org.elasticsearch.xpack.core.async.AsyncExecutionIdTests.randomAsyncId;
+
+public class GetAsyncStatusRequestTests extends AbstractWireSerializingTestCase<GetAsyncStatusRequest> {
+    @Override
+    protected Writeable.Reader<GetAsyncStatusRequest> instanceReader() {
+        return GetAsyncStatusRequest::new;
+    }
+
+    @Override
+    protected GetAsyncStatusRequest createTestInstance() {
+        GetAsyncStatusRequest req = new GetAsyncStatusRequest(randomSearchId());
+        if (randomBoolean()) {
+            req.setKeepAlive(TimeValue.timeValueMillis(randomIntBetween(1, 10000)));
+        }
+        return req;
+    }
+
+    @Override
+    protected GetAsyncStatusRequest mutateInstance(GetAsyncStatusRequest instance) {
+        return null;// TODO implement https://github.com/elastic/elasticsearch/issues/25929
```

Comment:
this should not be too difficult to implement for this specific request?

Reply:
OK, I'll try. I implemented it on the ResolveClusterActionRequest test and it was wonky and gave strange errors, but I forget the details. If I hit problems I'll document it here.

Reply:
Yeah, same as with the ResolveClusterActionRequestTest I did in another open PR - it always fails the `testEqualsAndHashcode` method.

```
java.lang.AssertionError: GetAsyncStatusRequest mutation should not be equal to original
Expected: not <org.elasticsearch.xpack.core.async.GetAsyncStatusRequest/unset>
     but: was <org.elasticsearch.xpack.core.async.GetAsyncStatusRequest/unset>
Expected :not <org.elasticsearch.xpack.core.async.GetAsyncStatusRequest/unset>
Actual   :<org.elasticsearch.xpack.core.async.GetAsyncStatusRequest/unset>
```

I spent a fair bit of time trying to figure out why that was happening for `ResolveClusterActionRequestTest` but gave up and since most other tests just marked that method as TODO and return null, I followed suit.

I ran it 20 times. It failed each time with the above error message.

Here's the implementation I tried:

```java
    @Override
    protected GetAsyncStatusRequest mutateInstance(GetAsyncStatusRequest instance) {
        if (instance.getKeepAlive() != null && randomBoolean()) {
            instance.setKeepAlive(null);
        } else {
            instance.setKeepAlive(new TimeValue(randomLongBetween(0, 9999999), TimeUnit.SECONDS));
        }
        return instance;
    }
```

Reply:
thanks a lot!